File: string/647.go
func countSubstrings(s string) int {
    var ret int
    for index := 0; index < len(s); index++ {
        for i, j := index, index+1; i >= 0 && j < len(s); i, j = i-1, j+1 {
            if s[i] == s[j] {
                ret++
            }else {
                break
            }
        }
        ret++
        for i, j := index-1, index+1; i >= 0 && j < len(s); i, j = i-1, j+1 {
            if s[i] == s[j] {
                ret++
            }else {
                break
            }
        }
    }
    return ret
}

File: string/647.cc
class Solution {
public:
    int countSubstrings(string s) {
        int ret = 0;
        for (int index = 0; index < s.size(); ++index) {
            for (int i = index, j = index+1; i >= 0 && j < s.size(); --i, ++j) {
                if (s[i] == s[j]) {
                    ++ret;
                } else {
                    break;
                }
            }
            ++ret;
            for (int i = index-1, j = index+1; i >= 0 && j < s.size(); --i, ++j) {
                if (s[i] == s[j]) {
                    ++ret;
                } else {
                    break;
                }
            }
        }
        return ret;
    }
};
